### DISCUSSION TITLE
v3.1.0

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#7923] [Releases & Announcements]
## Changelog

- Enhanced Knowledge Portal Integration with Chatwidget
- Advanced Email Message Signatures Supporting Image Attachments
- Optimized Contact Import Functionality
- Upgraded Slack Integration to Support Attachments
- Comprehensive Bug Resolutions and Performance Enhancements

Thanks to @jordan-brough, @LiamAshdown , @paulocr,  @thorakna, @gitstart, @BikashSah999   for the contributions

<hr /><em>This discussion was created from the release <a href='https://github.com/chatwoot/chatwoot/releases/tag/v3.1.0'>v3.1.0</a>.</em>

-----
## Note: 
The node version should be 20.x starting from Chatwoot version 3.1.0. Make sure to upgrade cwctl before running `--upgrade`.

1.  Upgrade cwctl via https://www.chatwoot.com/docs/self-hosted/deployment/linux-vm#install-or-upgrade-chatwoot-cli
2. Run Chatwoot upgrade with

```
sudo cwctl --upgrade
```

**Comment by HermesMacedo** [comment]
@sojan-official contact and conversation attribute filters do not work in automation;
#7931 https://github.com/chatwoot/chatwoot/issues/7931

**Comment by vishnu-narayanan** [comment]
The node version should be 20.x starting from Chatwoot version 3.1.0. Make sure to upgrade cwctl before running `--upgrade`.

1.  Upgrade cwctl via https://www.chatwoot.com/docs/self-hosted/deployment/linux-vm#install-or-upgrade-chatwoot-cli
2. Run Chatwoot upgrade with

```
sudo cwctl --upgrade
```

**Reply by fsentinel** [reply]
After upgrade, nothing change, only version information v3.2 , need  something else?

**Comment by ghost** [comment]
Hey, I'm facing a problem that I do not understand, since I did not made any changes in the files:

```cmd
sudo cwctl -u
Upgrading Chatwoot to v3.1.1
app/javascript/shared/constants/messages.js: needs update
public/assets/administrate/bot/avatar.png: needs update
public/assets/administrate/user/avatar.png: needs update
public/assets/images/auth/auth--bg.svg: needs update
public/assets/images/auth/bottom-right.svg: needs update
public/assets/images/auth/google.svg: needs update
public/assets/images/auth/top-left.svg: needs update
public/assets/images/dashboard/channels/api.png: needs update
public/assets/images/dashboard/channels/email.png: needs update
public/assets/images/dashboard/channels/facebook.png: needs update
public/assets/images/dashboard/channels/line.png: needs update
public/assets/images/dashboard/channels/microsoft.png: needs update
public/assets/images/dashboard/channels/sms.png: needs update
public/assets/images/dashboard/channels/telegram.png: needs update
public/assets/images/dashboard/channels/twilio.png: needs update
public/assets/images/dashboard/channels/twitter.png: needs update
public/assets/images/dashboard/channels/website.png: needs update
public/assets/images/dashboard/channels/whatsapp.png: needs update
public/assets/images/dashboard/editor/cmd-editor.png: needs update
public/assets/images/dashboard/editor/enter-editor.png: needs update
public/assets/images/dashboard/priority/high.svg: needs update
public/assets/images/dashboard/priority/low.svg: needs update
public/assets/images/dashboard/priority/medium.svg: needs update
public/assets/images/dashboard/priority/none.svg: needs update
public/assets/images/dashboard/priority/urgent.svg: needs update
public/assets/images/hc/grid.svg: needs update
Custom code changes detected. Aborting update.
Please proceed to update manually.
```

**Reply by vishnu-narayanan** [reply]
Which version are you upgrading from?

**Reply by ghost** [reply]
Hey! From 3.0.0 to 3.1.1.

**Comment by fsentinel** [comment]
Hi , I am on version 2.18 in production enviroment, update to 3.2? or wait for update with v3 stable? Thanks
